cscs-checks/cuda/cuda_memtest_check.py

```diff
@@ -0,0 +1,46 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.simple_test
+class FlexibleCudaMemtest(rfm.RegressionTest):
+    def __init__(self):
+        super().__init__()
+        self.valid_systems = ['daint:gpu', 'dom:gpu']
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
+                                    'PrgEnv-intel', 'PrgEnv-pgi']
+        self.descr = 'Flexible Cuda Memtest'
+        self.executable = 'osu_alltoall'
+        self.maintainers = ['TM', 'VK']
+        self.tags = {'maintenance'}
+        self.num_tasks_per_node = 1
+        self.num_tasks = 0
+        self.num_gpus_per_node = 1
+        self.modules = ['cudatoolkit']
+        self.sourcesdir = None
+        src_url = ('https://downloads.sourceforge.net/project/cudagpumemtest/'
+                   'cuda_memtest-1.2.3.tar.gz')
+        self.prebuild_cmd = ['wget %s' % src_url,
+                             'tar -xzf cuda_memtest-1.2.3.tar.gz '
+                             '--strip-components=1']
+        self.executable = 'cuda_memtest_sm20'
+        self.executable_opts = ['--disable_test', '6', '--num_passes', '1']
+
+        valid_test_ids = [i for i in range(11) if i not in {6, 9}]
+        assert_finished_tests = [sn.assert_eq(sn.count(sn.findall(
+            'Test%s finished' % test_id, self.stdout)),
+            self.num_tasks_assigned) for test_id in valid_test_ids]
```

Comment:
This expression is a bit hard to follow. Can you format a bit better, so as to be more readable?